### PR TITLE
Add query history sorting for remote queries

### LINKED ISSUE #1204: Show Query Directory opens parent folder
The new Show Query Directory dropdown is very useful. However, (at least on my system - Ubuntu 21.10) it opens the parent of the query directory, with the query directory selected. It would be nicer if it opened the actual query directory, since otherwise clicking away from the file browser makes it hard to get back to the right directory.

(Very minor and non-blocking)

### PR DESCRIPTION
**Pull request by aeisenberg**
Also, fix two smaller issues:

- Ensure the `Open Query Directory` command opens inside the specified
  directory.
- Ensure label changes are saved across restarts.

Fixes #1204

## Checklist

- [x] [CHANGELOG.md](https://github.com/github/vscode-codeql/blob/main/extensions/ql-vscode/CHANGELOG.md) has been updated to incorporate all user visible changes made by this pull request.
- [x] Issues have been created for any UI or other user-facing changes made by this pull request.
- [ ] _[Maintainers only]_ If this pull request makes user-facing changes that require documentation changes, open a corresponding docs pull request in the [github/codeql](https://github.com/github/codeql/tree/main/docs/codeql/codeql-for-visual-studio-code) repo and add the `ready-for-doc-review` label there.

### CODE REVIEW

File: extensions/ql-vscode/CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,8 @@
 ## [UNRELEASED]
 
 - Fix a bug where the AST viewer was not synchronizing its selected node when the editor selection changes. [#1230](https://github.com/github/vscode-codeql/pull/1230)
+- Open the directory in the finder/explorer (instead of just highlighting it) when running the command "Open query directory" command from the query history view. [#1235](https://github.com/github/vscode-codeql/pull/1235)
+- Ensure query label changes are persisted across restarts. [#1235](https://github.com/github/vscode-codeql/pull/1235)
```
Comment (shati-patel) [inline extensions/ql-vscode/CHANGELOG.md:7]: Maybe we can clarify what these labels are 💡 Something like this? 
```suggestion
- Ensure query label changes in the query history view are persisted across restarts. [#1235](https://github.com/github/vscode-codeql/pull/1235)
```

File: extensions/ql-vscode/src/query-history.ts [hunk 2]
```diff
@@ -727,20 +737,20 @@ export class QueryHistoryManager extends DisposableObject {
...
-    let p: string | undefined;
+    let externalFilePath: string | undefined;
     if (finalSingleItem.t === 'local') {
       if (finalSingleItem.completedQuery) {
-        p = finalSingleItem.completedQuery.query.querySaveDir;
+        externalFilePath = path.join(finalSingleItem.completedQuery.query.querySaveDir, 'timestamp');
       }
     } else if (finalSingleItem.t === 'remote') {
-      p = path.join(this.queryStorageDir, finalSingleItem.queryId);
+      externalFilePath = path.join(this.queryStorageDir, finalSingleItem.queryId, 'timestamp');
```
Comment (shati-patel) [inline extensions/ql-vscode/src/query-history.ts:746]: Nice use of the `timestamp` file! ⏲️

File: extensions/ql-vscode/src/query-history.ts [hunk 3]
```diff
@@ -727,20 +737,20 @@ export class QueryHistoryManager extends DisposableObject {
...
     }
 
-    if (p) {
+    if (externalFilePath) {
```
Comment (shati-patel) [inline extensions/ql-vscode/src/query-history.ts:749]: This is probably a fringe case, but is it worth also throwing a `Failed to open` error when this path doesn't exist (e.g. if a user has deleted the timestamp for some reason)?
At the moment it fails silently.

**Review by shati-patel — APPROVED** [review]
Thanks, this looks good to me! Just a few minor comments 🎉

File: extensions/ql-vscode/src/query-history.ts (hunk 3 above)
Comment (aeisenberg) [reply extensions/ql-vscode/src/query-history.ts:749]: Hmmm...unfortunately, the `revealFileInOS` command does not throw an error if the file doesn't exist. I guess I can add a `fs.pathExists` check before trying to open.

Comment (aeisenberg) [reply extensions/ql-vscode/src/query-history.ts:750]: Done

**Comment by aeisenberg** [comment]
@shati-patel can you take another look and see if your comment is addressed?

**Review by shati-patel — APPROVED** [review]
Thanks for the changes! LGTM 🙌🏽

**Pull request merged**